### PR TITLE
feat: add Value type Transformer

### PR DESCRIPTION
**Pull request by CurryPseudo**
only support native transformer now

### CODE REVIEW

File: src/values.rs [hunk 1]
```diff
@@ -660,3 +671,46 @@ fn check_division_by_zero(num: i32) -> Result<()> {
...
+impl Display for TransformerValue {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            TransformerValue::Native(_) => write!(f, "<build-in transformer>"),
+        }
+    }
+}
+
+impl TransformerValue {
+    pub fn transform_to_statements(&self, expressions: Vec<Expression>) -> Vec<Statement> {
```
Comment (Danielmelody) [inline src/values.rs:689]: expressions => expression_arguments
Vec => IntoIterator

File: src/values.rs [hunk 2]
```diff
@@ -660,3 +671,46 @@ fn check_division_by_zero(num: i32) -> Result<()> {
...
+        match self {
+            TransformerValue::Native(f) => f(expressions),
+        }
+    }
+    pub fn transform_to_expression(
```
Comment (Danielmelody) [inline src/values.rs:694]: This function should placed outside the transformer, or just add an `expect_expression` in `Statement`

File: src/values.rs (hunk 1 above)
Comment (Danielmelody) [inline src/values.rs:689]: just name it `transform`

**Review by Danielmelody — CHANGES_REQUESTED** [review]

File: src/values.rs [hunk 3]
```diff
@@ -660,3 +671,46 @@ fn check_division_by_zero(num: i32) -> Result<()> {
         _ => Ok(()),
     }
 }
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum TransformerValue {
```
Comment (Danielmelody) [inline src/values.rs:676]: Hmmm... it is common to name it `Transformer`, just the same as the variant name `Value::Transformer`

Comment (CurryPseudo) [reply src/values.rs:676]: there is a also a `Transformer` in parser/macro.rs, is it ok to keep this Value variant sturct and AST struct having same name `Transfomer`?

Comment (Danielmelody) [reply src/values.rs:676]: Yes, they are in different mods

**Comment by CurryPseudo** [comment]
updated

**Comment by CurryPseudo** [comment]
updated

**Pull request closed without merging**